src/js_app.hpp

```diff
@@ -160,18 +171,23 @@ void AppClass<T>::constructor(ContextType ctx, ObjectType this_object, Arguments
         throw std::runtime_error("Expected either a configuration object or an app id string.");
     }
 
-    Protected<typename T::GlobalContext> protected_ctx(Context<T>::get_global_context(ctx));
-    config.transport_generator = [=] {
-        return std::make_unique<NetworkTransport>(protected_ctx);
+    auto realm_constructor = js::Value<T>::validated_to_object(ctx, js::Object<T>::get_global(ctx, "Realm"));
+    
+    Protected<typename T::GlobalContext> protected_ctx(Context::get_global_context(ctx));
+    config.transport_generator = [protected_ctx] {
+        return AppClass<T>::transport_generator(protected_ctx);
     };
 
-    auto realm_constructor = js::Value<T>::validated_to_object(ctx, js::Object<T>::get_global(ctx, "Realm"));
     std::string user_agent_binding_info;
     auto user_agent_function = js::Object<T>::get_property(ctx, realm_constructor, "_createUserAgentDescription");
     if (js::Value<T>::is_function(ctx, user_agent_function)) {
         auto result = js::Function<T>::call(ctx, js::Value<T>::to_function(ctx, user_agent_function), realm_constructor, 0, nullptr);
         user_agent_binding_info = js::Value<T>::validated_to_string(ctx, result);
     }
+    else {
+        // FIXME: When debugging RN apps, _createUserAgentDescription cannot be found
+        user_agent_binding_info = "N/A";
```

Comment:
```suggestion
        user_agent_binding_info = "RN debugging";
```

Reply:
I understand that this would be the case from a practical level, but if we had a bug in our extension code the `_createUserAgentDescription` method might not be assigned and then I think it's misleading to send "RN debugging".

Reply:
I think it's a bit pessimistic to optimize for the case where we have bugs in our code. We should instead assume that the code works as intended and provide the most relevant UX to users.

Reply:
> I think it's a bit pessimistic to optimize for the case where we have bugs in our code.

I disagree.

The suggested change is too specific to the situations that might be causing the function to be undefined. In my opinion, the "right" solution would be to provide an authoritative runtime check which determines is we're running in the chrome debugger mode, but I honestly see very little value in that, since these values likely wont be relevant to the developer through the MongoDB Realm admin dashboard.

Reply:
I agree that an unambiguous check if the code is running in a debugger is better, but don't agree with the rest. I don't think a user will easily make the connection that "js-unknown" means a debugger, so is likely to be left with the impression that either our code or theirs is broken and doesn't correctly report the platform.

If this function is only unrefined in a debugger, then using that as a check, even if semantically incorrect, is better than providing a broken user experience.

Reply:
In pre-v10 we skipped setting it as object store didn't require it. With v10, it is now a requirement. You can say that we are doing more or less the same as before.